tests/test_Voting.py

```diff
@@ -0,0 +1,352 @@
+    dpytest.verify_message(
+        f"Vote titled `Test Vote` created for guild {guild.name}. Use `{KoalaBot.COMMAND_PREFIX}help vote` to see how to configure it.")
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote addRole {guild.roles[0].id}")
+    dpytest.verify_message(f"Vote will be sent to those with the {guild.roles[0].name} role")
+    vote = cog.vote_manager.get_configuring_vote(guild.members[0].id)
+    assert guild.roles[0].id in vote.target_roles
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote removeRole {guild.roles[0].id}")
+    dpytest.verify_message(f"Vote will no longer be sent to those with the {guild.roles[0].name} role")
+    assert guild.roles[0].id not in vote.target_roles
+
+
+@pytest.mark.asyncio
+async def test_discord_set_chair():
+    config = dpytest.get_config()
+    guild = config.guilds[0]
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote create Test Vote")
+    dpytest.verify_message(
+        f"Vote titled `Test Vote` created for guild {guild.name}. Use `{KoalaBot.COMMAND_PREFIX}help vote` to see how to configure it.")
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote setChair {guild.members[0].id}")
+    dpytest.verify_message(f"Set chair to {guild.members[0].name}")
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote setChair")
+    dpytest.verify_message("Results will be sent to the channel vote is closed in")
+
+@pytest.mark.asyncio
+async def test_discord_add_remove_option():
+    config = dpytest.get_config()
+    guild = config.guilds[0]
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote create Test Vote")
+    dpytest.verify_message(
+        f"Vote titled `Test Vote` created for guild {guild.name}. Use `{KoalaBot.COMMAND_PREFIX}help vote` to see how to configure it.")
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote addOption test+test")
+    dpytest.verify_message("Option test with description test added to vote")
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote addOption testtest")
+    dpytest.verify_message("Example usage: k!vote addOption option title+option description")
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote removeOption 1")
+    dpytest.verify_message("Option number 1 removed")
+
+
+@pytest.mark.asyncio
+async def test_discord_cancel_vote():
+    config = dpytest.get_config()
+    guild = config.guilds[0]
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote create Test Vote")
+    dpytest.verify_message(
+        f"Vote titled `Test Vote` created for guild {guild.name}. Use `{KoalaBot.COMMAND_PREFIX}help vote` to see how to configure it.")
+    await dpytest.message(f"{KoalaBot.COMMAND_PREFIX}vote cancel Test Vote")
+    dpytest.verify_message("Vote Test Vote has been cancelled.")
+
+# can't get any tests that involve sending embeds to work
+@pytest.mark.skip(reason="can't get any tests that involve sending embeds to work")
```

Comment:
Do you know why sending embeds doesn't work for testing?

Reply:
reactions aren't implemented in dpytest, so really it's that issue afiak

Reply:
would it be worth in a later stage to fork dpytest to implement reactions?